### ISSUE TITLE
ModernMT refuses to train new engines

### ISSUE THREAD

**Issue by LoekvanKooten** [modernmt/modernmt#559]
No matter what I try, suddenly ModernMT only gives me this. Runtime folder empty. Engine folder empty. Restart doesn't change anything.

Also, ModernMT constantly crashes when training on two GPU's. And no, it's not temperature-related. Power Supply also is 1000 Wat..

`ERROR Unexpected exception: Command 'fairseq-train /home/loek/modernmt4/runtime/procdut/tmp/training/data_generated --save-dir /home/loek/modernmt4/runtime/procdut/tmp/training/_temp_train/nn_model --task mmt_translation --user-dir /home/loek/modernmt4/build/mmt-4.5.4.jar/mmt --share-all-embeddings --no-progress-bar --tensorboard-logdir /home/loek/modernmt4/runtime/procdut/tmp/training/_temp_train/tensorboard_logdir --dataset-impl mmap --gpus1 --arch transformer_mmt_base --clip-norm 0.0 --label-smoothing 0.1 --attention-dropout 0.1 --dropout 0.3 --weight-decay 0.0 --criterion label_smoothed_cross_entropy --optimizer adam --adam-betas (0.9, 0.98) --log-interval 100 --lr 0.0005 --lr-scheduler inverse_sqrt --min-lr 1e-09 --warmup-init-lr 1e-07 --warmup-updates 4000 --max-tokens 3072 --update-freq 4 --save-interval-updates 1000 --keep-interval-updates 10 --keep-last-epochs 10' failed with exit code 2`

**Comment by LoekvanKooten** [comment]
Entering the fairseq command gives:

`bash: syntax error near unexpected token '('`

**Comment by nicolabertoldi** [comment]
HI @LoekvanKooten 

It happens sometimes that a particular batch, whose content and size depend on several parameters, lets the system crash.
Usually, the easiest solution is to modify the value of the''--max-tokens'' parameter (for instance from 1536 to 1472).

We are working for a more robust way to solve this issue.
It could be useful for us to have the full training.log reporting the failure.

thanks for your feedback

**Comment by LoekvanKooten** [comment]
Yes, a solution would be great. I'm now training another engine for another language pair, and this one crashes straight after 20 seconds of training or so. The crash times are always very consistent. With one dataset it's after 20 seconds, with another dataset after 2 hours, but always after 2 hours.

As for now I'll just let he 2nd graphics card sit and relax using --gpus 0 But this does mean much slower training times.
